### PR TITLE
feat(ui): consistent user dropdown for owner and reviewers

### PR DESCRIPTION
**Pull request by chirag-madlani**
### Describe your changes :

I worked on the create common component to show review and owner selection component

Closes: #10302 

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :

1. User selector
Screenshot 2023-03-16 at 3 47 35 AM

2. User or Team Selector
Screenshot 2023-03-16 at 10 46 08 AM

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/yarn.lock
Comment (ShaileshParmar11) [inline openmetadata-ui/src/main/resources/ui/yarn.lock:1]: is it intentional to puch a change for yarn.lock?

Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/yarn.lock:1]: Yes it's to remove warning present while doing yarn install

File: openmetadata-ui/src/main/resources/ui/src/styles/variables.less [hunk 1]
```diff
@@ -21,10 +21,10 @@
 @failed-color: #cb2431;
 @aborted-color: #efae2f;
 @info-color: #1890ff;
-@text-color: #000000;
+@text-color: #37352f;
 @text-color-secondary: #37352f;
 @text-color-tertiary: #00000080;
-@text-grey-muted: #6b7280;
+@text-grey-muted: #6b7280d9;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/styles/variables.less:27]: Do we want to add alpha value as well?

File: openmetadata-ui/src/main/resources/ui/src/components/common/UserSelectableList/UserSelectableList.interface.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    } & (
+      | {
+          multiSelect?: true;
+          onUpdate: (updatedUsers: EntityReference[]) => void;
+        }
+      | {
+          multiSelect: false;
+          onUpdate: (updatedUsers: EntityReference) => void;
+        }
+    );
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/UserSelectableList/UserSelectableList.interface.ts:32]:
```suggestion
export type UserSelectableListProps {
  hasPermission: boolean;
  selectedUsers: EntityReference[];
  onUpdate: (updatedUsers: EntityReference | EntityReference[]) => void;
  multiSelect?: boolean;
  children?: ReactNode;
  popoverProps?: PopoverProps;
}
```

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

File: openmetadata-ui/src/main/resources/ui/src/components/common/EntitySummaryDetails/EntitySummaryDetails.tsx [hunk 3]
```diff
@@ -86,13 +87,28 @@ const EntitySummaryDetails = ({
...
+
   const [showTypeSelector, setShowTypeSelector] = useState(false);
 
   const handleShowTypeSelector = useCallback((value: boolean) => {
     setShowTypeSelector(value);
   }, []);
 
+  const ownerDropdown = allowTeamOwner ? (
+    <UserTeamSelectableList
+      hasPermission={Boolean(updateOwner)}
```
[the remark this answers is not in the document]
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/common/EntitySummaryDetails/EntitySummaryDetails.tsx:98]: Yes

File: openmetadata-ui/src/main/resources/ui/src/styles/variables.less (hunk 1 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/styles/variables.less:27]: Yes

File: openmetadata-ui/src/main/resources/ui/src/components/common/UserSelectableList/UserSelectableList.interface.ts (hunk 2 above)
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/src/components/common/UserSelectableList/UserSelectableList.interface.ts:32]: It won't work for conditional props restriction on parent level

File: openmetadata-ui/src/main/resources/ui/cypress/e2e/Flow/AddTeamAsOwner.spec.js [hunk 4]
```diff
@@ -70,17 +70,17 @@ describe('Create a team and add that team as a owner of the entity', () => {
 
     interceptURL(
       'GET',
-      '/api/v1/search/query?q=*%20AND%20teamType:Group&from=0&size=10&index=team_search_index',
+      '/api/v1/search/query?q=*%20AND%20teamType:Group&from=0&size=15&index=team_search_index',
```
[the remark this answers is not in the document]
Comment (chirag-madlani) [reply openmetadata-ui/src/main/resources/ui/cypress/e2e/Flow/AddTeamAsOwner.spec.js:73]: Will do it later

**Review by ShaileshParmar11 — DISMISSED** [review]

**Review by harshach — APPROVED** [review]

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**